nativelib/src/main/scala/java/net/SocketHelpers.scala

```diff
@@ -65,15 +65,19 @@ private[net] object SocketHelpers {
         } 
 
         if (select(sock + 1, fdset, null, null, time) != 1) {
-          return false;
+          return false
         } else {
           val buf      = stackalloc[CChar](5)
           val recBytes = recv(sock, buf, 5, 0)
           if (recBytes < 4) {
             return false
           }
         }
-      } finally {
+      } 
+      catch {
```

Comment:
I've encountered a weird bug here - it doesn't work without the `catch` clause. If I try to compile with just `try-finally`, the compiler throws a NullPointerException. I was not able to reproduce it on simple `try` blocks, though I will try to minimize the reproduction and file an issue with it.

The catch clause doesn't affect the logic of the code, so I guess it can be fixed later.